docs/guides/caddy-configuration.md

```diff
@@ -46,17 +50,45 @@ blackhole:80, pi.hole:80, 0.0.0.0:80 {
 
 In this case, I've chosen to also add blackhole and pi.hole as valid names to open the admin page with.
 
-Finally, restart your Caddy server: `sudo service caddy restart`
+### Caddyfile (Caddy version 2)
+
+```
+pi.hole:80{
+  reverse_proxy localhost:1080
+}
+```
+
+- If you'd like to enable Https on your site, make sure your server is reacheable via your domain name (ex: myawesomesite.com) and is pointing to the right IP address.
+- Additionally you need to open ports :80 and :443 (Apart from the one's required specifically for pi-hole) for your server before setting up https.
+
+The following configuration will automatically fetch and setup Https for your domain using Lets-Encrypt
+
+```
+myawesomesite.com {
+  reverse_proxy localhost:1080
+}
+```
+
+Additionally you can make pihole reacheable via a subdomain and optionally can you enable Zstandard and Gzip compression as follows:
+
+```
+pihole.myawesomesite.com {
+  reverse_proxy localhost:1080
+  encode zstd gzip
+}
+```
+
+Finally, run `sudo systemctl daemon-reload` to force caddy to load the new configuration.
```

Comment:
Also this won't reload Caddy, it'll only reload systemd. The correct command would be `sudo systemctl reload caddy` if the service was correctly set up.